Step 1 GitOps changes (#2317)
giantswarm/docs: src/content/overview/fleet-management/app-management/index.md

---
linkTitle: App management
title: App management
description: The app platform allows to manage app catalogs and apps, for simple and standardized deployment across the platform.
weight: 40
menu:
  principal:
    parent: overview-fleet-management
    identifier: overview-fleet-management-app-platform
owner:
  - https://github.com/orgs/giantswarm/teams/team-honeybadger
user_questions:
  - What does App Platform do?
  - What is an App Catalog?
  - What is a Managed App?
  - Does Giant Swarm provide app catalogs out of the box?
  - How is App Platform implemented?
  - How does the App Platform relate to Helm?
  - What are the components of App Platform?
  - What are the app catalogs that Giant Swarm provides out of the box?
  - How can I create an organizational app catalog?
  - How can I interact with the Giant Swarm App Platform?
  - Can I create an app catalog?
last_review_date: 2024-10-22
---

> The _Giant Swarm App Platform_ refers to a set of features and concepts that allow you to browse, install and manage the configurations of apps (such as `Prometheus`) from a single place; the [platform API]({{< relref "/overview/architecture#platform-api" >}}).

Reviewer: There is a mixture of casing of app names when enclosed as inline code that feels off. Here you use `Prometheus`, in the next paragraph `Helm`, yet lower down in the doc `coredns` and `cluster-autoscaler` are referred as lowercase variants.

Generally I am of the opinion that app names should be lowercase when referred to as inline code, reflecting the casing you would use as part of a command, however opinions may vary but we should be consistent in the casing we choose.

Author: We agreed on using lowercase; sadly, I have trouble finding and replacing it. Looking at this

> Giant Swarm fully support [`Helm`](https://helm.sh/) as a general tool to deploy your applications as well as for our general `App Catalog`. Apps are packaged as `Helm` charts and can be configured with _values_. A recommended [app configuration]({{< relref "/tutorials/app-platform/app-configuration" >}}) is provided which you can override to meet your needs.

Reviewer: ```suggestion
Giant Swarm fully supports [`Helm`](https://helm.sh/) as a general tool to deploy your applications as well as for our `App Catalog`. Apps are packaged as `Helm` charts and can be configured with _values_. A recommended [app configuration]({{< relref "/tutorials/app-platform/app-configuration" >}}) is provided which you can override to meet your needs.
```

Duplication of the word `general` - the second is definitely redundant, 

`support` is a verb in this context, Giant Swarm is singular, "we support", "it supports"

> The app platform then, underneath, installs these `Helm` Charts whenever an app installation is requested by you. The `Helm` execution is mostly not configurable, with the exception to the options presented in

Reviewer: ```suggestion
The app platform installs these `Helm` Charts whenever an app installation is requested by you. The `Helm` execution is mostly not configurable, with the exception to the options presented in
```

`then, underneath` is redundant and confusing, I would leave this out.

[installation configuration]({{< relref "/tutorials/app-platform/installation-configuration" >}}).

> This feature of the platform provides a collection of curated managed apps. These managed apps are grouped into app catalogs, which can be browsed through our web interface. They may also install their own catalogs using the platform API. Finally, it's worth noting that Giant Swarm uses the app platform to install the pre-installed apps in your clusters, such as `coredns` or `cluster-autoscaler`.

Reviewer: ```suggestion
This feature of the platform provides a collection of curated managed apps. These managed apps are grouped into app catalogs, which can be browsed through our web interface. They may also install their own catalogs using the platform API. Finally, it's worth noting that Giant Swarm uses the app platform to pre-install apps such as `coredns` or `cluster-autoscaler` into your clusters.
```

- Version >= `v1.0.0` with no suffix - the specified version of the application is at a stable maturity level. It's available to our customers as a managed offering with support and SLA.

### Managed app definition

A `managed app` is an app in our Giant Swarm catalog that provides:

- Safe and tested deployment

The `Helm` chart is ready to use and tested, either sanitizing the upstream forked applications or creating good defaults for our maintained ones. [It's offered a common way of building apps](https://github.com/giantswarm/app-build-suite) and a [testing framework](https://github.com/giantswarm/app-test-suite) which ensure the application is deployable and works as expected. Security and upgrade viability are checked too during the integration process.

- Monitoring

Giant Swarm makes sure all the main components of the app are running and that the app is working as expected. At the same time, our automation set up monitoring and alerting on necessary metrics to ensure our service level agreements.

In case of an alert, operations team perform an Root Cause Analysis (RCA) to understand if it's a Giant Swarm or customer-inflicted issue that broke the application.

> __Note__: Giant Swarm don't fix bugs upstream when they involve significant code changes. That said, our team always try to find a workaround or the root cause of the issue and submit a ticket to the upstream project. In some cases, our engineers fix the bug ourselves where it’s necessary and possible for us, and provide the fix to the upstream project. This might result in temporarily running a non-upstream patch version from Giant Swarm until upstream merges our patch. Customers can, in general, expect the same level of service for a managed optional app as they get with "default" apps such as `coredns`.

Reviewer: ```suggestion
__Note__: In general, Giant Swarm does not fix bugs in upstream projects, however our team will always try to find a solution to an issue, or the root cause of the issue and submit a ticket to the upstream project when relevant to do so. In certain instances, our engineers may attempt to fix the bug where it is necessary and possible for us, and provide the fix to the upstream project. This can result in Giant Swarm temporarily running an intermediate patch version whilst waiting for upstream to release a new version containing our patch. Customers can expect the same level of service for a managed optional app as they get with "default" apps such as `coredns`.
```
